src/main/java/de/zalando/spring/cloud/config/aws/kms/KmsTextEncryptor.java

```diff
@@ -72,15 +74,75 @@ public String decrypt(final String encryptedText) {
             return EMPTY_STRING;
         } else {
 
+            // Extract the encryption context and the remaining part
+            final Map<String, String> encryptionContext = extractEncryptionContext(encryptedText);
+            final String encryptedValue = extractEncryptedValue(encryptedText);
+
             // Assuming the encryptedText is encoded in Base64
-            final ByteBuffer encryptedBytes = ByteBuffer.wrap(Base64.decode(encryptedText.getBytes()));
+            final ByteBuffer encryptedBytes =
+                    ByteBuffer.wrap(Base64.decode(encryptedValue.getBytes()));
 
-            final DecryptRequest decryptRequest = new DecryptRequest().withCiphertextBlob(encryptedBytes);
+            final DecryptRequest decryptRequest = new DecryptRequest()
+                    .withCiphertextBlob(encryptedBytes)
+                    .withEncryptionContext(encryptionContext);
 
             return extractString(kms.decrypt(decryptRequest).getPlaintext());
         }
     }
 
+    /**
+     * Extract the encryption context. For a string such as "(param1=WdfaA,param2=AZrr,param3)thisIsEncrypted",
+     * the value returned would be a map with pairs of key and values, such as:
+     *
+     * <ul>
+     * <li>key=param1,value=WdfaA</li>
+     * <li>key=param2,value=AZrr</li>
+     * <li>key=param3,value=</li>
+     * </ul>
+     *
+     * <p>Keys with no value are assigned an empty string for convenience. The remaining values are Base64 encoded.</p>
+     *
+     * @param encryptedText the encrypted text passed by spring security
+     * @return the encryption context map, without the encryption text
+     */
+    /* default for testing */ static Map<String, String> extractEncryptionContext(final String encryptedText) {
+        final int lower = encryptedText.indexOf('(');
+        final int upper= encryptedText.indexOf(')');
+        if (lower != 0 || upper < 0) {
+            return Collections.emptyMap();
+        } else {
+            final Map<String, String> encryptionContext = new HashMap<>();
+            final String encryptionContextText = encryptedText.substring(lower+1, upper);
+            final String[] pairs = encryptionContextText.split(",");
```

Comment:
Not sure too. We could add some code to prevent these empty values, but given they are filtered out later, when we look for the = symbol, I'd say that it is probably better to leave this way and, if necessary, update the documentation with cases like this.